gdal/frmts/grass/grass57dataset.cpp

```diff
@@ -454,23 +416,23 @@ CPLErr GRASSRasterBand::ResetReading ( struct Cell_head *sNewWindow )
          sNewWindow->ew_res != sOpenWindow.ew_res || sNewWindow->ns_res != sOpenWindow.ns_res ||
          sNewWindow->rows   != sOpenWindow.rows   || sNewWindow->cols   != sOpenWindow.cols )
     {
-        SetWindow ( sNewWindow );
-        G_copy((void *) &sOpenWindow, (void *) sNewWindow, sizeof(struct Cell_head));
+        Rast_set_window ( sNewWindow );
+        memcpy((void *) &sOpenWindow, (void *) sNewWindow, sizeof(struct Cell_head));
     }
     else
     {
         /* The windows are identical, check current window */
         struct Cell_head sCurrentWindow;
 
-        G_get_window ( &sCurrentWindow );
+        Rast_get_window ( &sCurrentWindow );
 
         if ( sNewWindow->north  != sCurrentWindow.north  || sNewWindow->south  != sCurrentWindow.south ||
              sNewWindow->east   != sCurrentWindow.east   || sNewWindow->west   != sCurrentWindow.west ||
              sNewWindow->ew_res != sCurrentWindow.ew_res || sNewWindow->ns_res != sCurrentWindow.ns_res ||
              sNewWindow->rows   != sCurrentWindow.rows   || sNewWindow->cols   != sCurrentWindow.cols
              )
         {
-            SetWindow ( sNewWindow );
+            Rast_set_window ( sNewWindow );
```

Comment:
this change is wrong. The call to the C++ SetWindow() function is intended.

Reply:
Thanks, hope I got it right in the updated version (now: grass.cpp)

Reply:
You should check with @metzm who added SetWindow() very recently per 28a38e366efaa59aaad9cb074d2dbf8358aa9c13

Reply:
Now I got it thanks to this reference. Fixed and pytest passes again.